### ISSUE TITLE
[DSP] Don't Stop Pretraining: Adapt Language Models to Domains and Tasks

### ISSUE THREAD

**Issue by bigshanedogg** [bigshanedogg/survey#3]
## TL;DR:
> - The author argues that training model to recognize the relevance between context and response is important for response selection task.
> - The overall architecture seems similar with MDFN(Masking Decoupling Fusing Network) in that it fuses information of different level such as word, utterance, and context through attention mechanism.
> - The mlm loss after DAPT has been decreased except news, which indicates that DAPT is significant for more distant domain.

## Introduction
![figure_02](https://user-images.githubusercontent.com/18084680/180647506-0a67c056-4ecb-4098-9019-aef8728ae14d.png)
- Adapting pretrained model to downstream task are followed by several questions. One of them is how to benefit the pretrained model, since the pretrained corpus may be dissimilar with domain corpus.
- When there is difference between the pretrained corpus and domain corpus, the DAPT (domain adaptive training), continual pretraining on domain corpus, may improve the performance.
- This paper is experiment log of adapting DAPT and TAPT on various domain and task combination.

## DAPT & TAPT
- The purpose of The purpose of DAPT is overwriting the embedding space with domain corpus while not losing learned linguistic features from pretrained language model. Likewise, task adaptive training (TAPT) focused on performance improvement through continual pretraining on task dataset.
- The domains experimented in this paper are computer science(cs), biomedical(biomed), news, and movie reviews(reviews).
- TAPT focused on more relevant courpus, while exploiting smaller dataset even than DAPT.
- Therefore, DAPT would be more robust on other domain-specific tasks than TAPT since it is more resource intensive, although the efficiency on the task is lower than TAPT.

## Which performs the best?
- DAPT followed by TAPT showed the best performance. TAPT followed by DAPT was speculated that forget the task-relevant corpus by broader domain corpus.
- Adapting both DAPT and TAPT seems expensive at first glance, but the model pretrained with domain corpus (i.e. DAPT) can be reused to other in-domain task by adapting TAPT per task with small amount of dataset and time.

## Ablation
![figure_03](https://user-images.githubusercontent.com/18084680/180647517-d3026c2c-2e24-450c-9938-f8437085dd44.png)
- kNN-TAPT is TAPT on task dataset augmented with lightweight model to choose top-k similar sentences from domain dataset. The higher k, The more improved performance as intuition.

## Conclusion
- The mlm loss after DAPT has been decreased except news, which indicates that DAPT is significant for more distant domain. The author assumed that high similarity between corpora causes the marginal increase in news domain.
- The DAPT to relevant domain -> performance increased
- The DAPT to irrelevant domain -> performance decreased
- DAPT increased the performance on both high, low resource settings
- In conclusion, DAPT and TAPT both mitigate the gap between pretrained corpus and in-domain corpus, and improve the downstream performance. In addition, the fact that the best combination is DAPT followed by TAPT indicates that DAPT and TAPT complements each other by focusing task-specific information after learning domain knowledge.

### Reference and Implementation:
- paper: [Don't Stop Pretraining: Adapt Language Models to Domains and Tasks](https://arxiv.org/abs/2004.10964), 23 Apr 2020
